features/admin.actions.v1/components/action-config-form.tsx

```diff
@@ -0,0 +1,841 @@
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { SelectChangeEvent } from "@mui/material";
+import Alert from "@oxygen-ui/react/Alert";
+import AlertTitle from "@oxygen-ui/react/AlertTitle";
+import Box from "@oxygen-ui/react/Box";
+import Button from "@oxygen-ui/react/Button";
+import Divider from "@oxygen-ui/react/Divider";
+import InputAdornment from "@oxygen-ui/react/InputAdornment";
+import Skeleton from "@oxygen-ui/react/Skeleton";
+import { useRequiredScopes } from "@wso2is/access-control";
+import { AppState, FeatureConfigInterface } from "@wso2is/admin.core.v1";
+import { AlertLevels, IdentifiableComponentInterface, LoadableComponentInterface } from "@wso2is/core/models";
+import { addAlert } from "@wso2is/core/store";
+import { URLUtils } from "@wso2is/core/utils";
+import {
+    FinalForm,
+    FinalFormField,
+    FormRenderProps,
+    FormSpy,
+    SelectFieldAdapter,
+    TextFieldAdapter
+} from "@wso2is/form";
+import { EmphasizedSegment, Heading, Hint } from "@wso2is/react-components";
+import { AxiosError, AxiosResponse } from "axios";
+import React, { Fragment, FunctionComponent, ReactElement, useEffect, useState } from "react";
+import { Trans, useTranslation } from "react-i18next";
+import { useDispatch, useSelector } from "react-redux";
+import { Dispatch } from "redux";
+
+import { Icon } from "semantic-ui-react";
+import { KeyedMutator } from "swr";
+import createAction from "../api/create-action";
+import updateAction from "../api/update-action";
+import { ActionsConstants } from "../constants/actions-constants";
+import {
+    ActionConfigFormPropertyInterface,
+    ActionInterface,
+    ActionResponseInterface,
+    ActionUpdateInterface,
+    AuthenticationPropertiesInterface,
+    AuthenticationType,
+    AuthenticationTypeDropdownOption
+} from "../models/actions";
+import "./action-config-form.scss";
+
+/**
+ * Prop types for the action configuration form component.
+ */
+interface ActionConfigFormInterface extends IdentifiableComponentInterface, LoadableComponentInterface {
+    /**
+     * Action's initial values.
+     */
+    initialValues: ActionConfigFormPropertyInterface;
+    /**
+     * Action Type of the Action.
+     */
+    actionTypeApiPath: string;
+    /**
+     * Specifies action creation state.
+     */
+    isCreateFormState: boolean;
+    /**
+     * Mutate api.
+     */
+    mutate: KeyedMutator<AxiosResponse<ActionResponseInterface[]>>
```

Comment:
I'm +1 to pass down as a prop, since if the hook is used in this component as well it'll cause an unnecessary API call?
Rather update the prop name to something like, `onActionUpdated: () => void;` and handle mutate inside the parent component. 
@pavinduLakshan WDYT?

Reply:
Added pavindu's suggestion. Let me know if this needs to revert back to previous status

Reply:
> I'm +1 to pass down as a prop, since if the hook is used in this component as well it'll cause an unnecessary API call?

We can avoid that by passing a shouldFetch param and still use the hook to get the mutate function, can't we?